rules/manager.go

```diff
@@ -85,6 +85,37 @@ func DefaultEvalIterationFunc(ctx context.Context, g *Group, evalTimestamp time.
 	g.setEvaluationTime(timeSinceStart)
 	g.setLastEvaluation(start)
 	g.setLastEvalTimestamp(evalTimestamp)
+
+	if g.alertStore != nil {
+		// feature enabled.
+		go func() {
```

Comment:
Why is this on a separate goroutine?

Reply:
In order to store the state asynchronously without blocking the group evaluation.

Reply:
Is that safe?
What happens if it takes a long time?

Reply:
Tested by adding random sleeps and did not notice any performance/latency issues.
If it takes a long time, a risk with this approach is incomplete work/loss of state from latest evaluation if Prometheus terminates before this goroutine since we do not use a WaitGroup.

Reply:
I'm worried that these goroutines will pile up.
Putting it more plainly, you must not do this.

Reply:
I see, thanks for the feedback. Removed, rule evaluation duration does not change significantly even if this is done on the same goroutine.